### ISSUE TITLE
Cookie Monster for the Steam Workshop?

### ISSUE THREAD

**Issue by TFHatGuy** [CookieMonsterTeam/CookieMonster#940]
**Describe the function you'd like to see added to CookieMonster**
Since the Steam version of Cookie Clicker is set to include a modding workshop, and the fact that [the FAQ ](https://orteil42.tumblr.com/post/659056422845333504/so-ive-spent-the-past-day-reading-over-everyones) mentions that "the Steam version is an electron wrapper around the HTML game," are there any plans to port Cookie Monster to the upcoming Steam Workshop? This mod adds a ton of cool functions that I'd like to be able to play around with in the Steam version as soon as I'm able to.

**Comment by DanielNoord** [comment]
I think we will need to see how the modding workshop is set up before we can give an estimate about how much time it will take to port Cookie Monster to the Steam version.

I'm pinning this issue for future reference.

**Comment by TFHatGuy** [comment]
> 
> 
> I think we will need to see how the modding workshop is set up before we can give an estimate about how much time it will take to port Cookie Monster to the Steam version.
> 
> I'm pinning this issue for future reference.

It's great to hear that a Steam version of Cookie Monster is being planned, even if it may take a little while. With any hope, porting the mod over won't take much effort at all, and it can get out relatively soon. We'll have to see, but again, I'm glad a Steam version is probably in the cards.

**Comment by AFKoncore** [comment]
The Steam version already support local mods, it expects an info.txt and a main.js.
You could provide those for people who don't want to edit any file, even if it's pretty simple stuff.

**Comment by Urganot** [comment]
You can kind of get it to run by copying one sample project and changing info.txt to
`{
	"Name": "CookieMonster",
	"ID": "cookie monster",
	"Author": "CookieMonster",
	"Description": "CookieMonster.",
	"ModVersion": 1,
	"GameVersion": 2.04,
	"Date": "13/08/2021",
	"Dependencies": [],
	"Disabled": 0
}`

and main.js to 
`Game.registerMod("cookie monster",{
	init:function(){
		Game.LoadMod('https://cookiemonsterteam.github.io/CookieMonster/dist/CookieMonster.js');
	},
});
`

The only issue is that clicking on the cookie doesnt do anything

**Comment by TFHatGuy** [comment]
> 
> 
> You can kind of get it to run by copying one sample project and changing info.txt to
> `{ "Name": "CookieMonster", "ID": "cookie monster", "Author": "CookieMonster", "Description": "CookieMonster.", "ModVersion": 1, "GameVersion": 2.04, "Date": "13/08/2021", "Dependencies": [], "Disabled": 0 }`
> 
> and main.js to
> `Game.registerMod("cookie monster",{ init:function(){ Game.LoadMod('https://cookiemonsterteam.github.io/CookieMonster/dist/CookieMonster.js'); }, }); `
> 
> The only issue is that clicking on the cookie doesnt do anything

In addition to that, you may want to set "AllowSteamAchievs":1, just in case; in my experience, Cookie Monster doesn't break the game in such a way that Steam achievements shouldn't be awarded while using it, and mods prohibit Steam achievements by default.

**Comment by TFHatGuy** [comment]
> 
> 
> You can kind of get it to run by copying one sample project and changing info.txt to
> `{ "Name": "CookieMonster", "ID": "cookie monster", "Author": "CookieMonster", "Description": "CookieMonster.", "ModVersion": 1, "GameVersion": 2.04, "Date": "13/08/2021", "Dependencies": [], "Disabled": 0 }`
> 
> and main.js to
> `Game.registerMod("cookie monster",{ init:function(){ Game.LoadMod('https://cookiemonsterteam.github.io/CookieMonster/dist/CookieMonster.js'); }, }); `
> 
> The only issue is that clicking on the cookie doesnt do anything

Having had some time to try out making a local mod for Cookie Monster following these steps, I consistently have ran into this error when launching the game with Cookie Monster's local mod on. 
![image](https://user-images.githubusercontent.com/37784396/131762669-e7eea9cf-08fc-45b8-8fa1-0b85a042932c.png)

**Comment by LeXiMoLe97** [comment]
> > You can kind of get it to run by copying one sample project and changing info.txt to
> > `{ "Name": "CookieMonster", "ID": "cookie monster", "Author": "CookieMonster", "Description": "CookieMonster.", "ModVersion": 1, "GameVersion": 2.04, "Date": "13/08/2021", "Dependencies": [], "Disabled": 0 }`
> > and main.js to
> > `Game.registerMod("cookie monster",{ init:function(){ Game.LoadMod('https://cookiemonsterteam.github.io/CookieMonster/dist/CookieMonster.js'); }, }); `
> > The only issue is that clicking on the cookie doesnt do anything
> 
> Having had some time to try out making a local mod for Cookie Monster following these steps, I consistently have ran into this error when launching the game with Cookie Monster's local mod on.
> ![image](https://user-images.githubusercontent.com/37784396/131762669-e7eea9cf-08fc-45b8-8fa1-0b85a042932c.png)

I got that error because i wasnt formatting it right

 `{
	"Name": "Cookie Monster",
	"ID": "cookie monster",
	"Author": "CookieMonster",
	"Description": "CookieMonster",
	"ModVersion": 1,
	"GameVersion": 2.04,
	"Date": "01/09/2021",
	"Dependencies": [],
	"Disabled": 1,
	"AllowSteamAchievs": 1
}` 

This works for me

**Comment by DanielNoord** [comment]
Good to know! Thanks guys! 
Whenever I have time (and actually have the game) I will work on #950

**Comment by davidak** [comment]
The workshop is now available with the latest update!

**Comment by CorruptComputer** [comment]
With the workshop now available how do I install this? I can't seem to find any documentation on it.

**Comment by 3plus4i** [comment]
someone already put Cookie Monster in the workshop, I don't think it's official though

**Comment by DanielNoord** [comment]
@3plus4i That is indeed true! However, as long as the steam version is Windows-only I can't do anything to support it myself as I can't test anything.

If the uploader of the mod would like to get into contact me they can do so here 😄

**Comment by Wacyrn** [comment]
Actually Cookie Monster has put several times to steam workshop already. there are 5 mods named Cooke Monster:
https://steamcommunity.com/sharedfiles/filedetails/?id=2685721341
https://steamcommunity.com/sharedfiles/filedetails/?id=2703796259
https://steamcommunity.com/sharedfiles/filedetails/?id=2687907415
https://steamcommunity.com/sharedfiles/filedetails/?id=2687907415
https://steamcommunity.com/sharedfiles/filedetails/?id=2696645125

**Comment by gusluchetti** [comment]
@DanielNoord Hey, I added the dev version of Cookie Monster to the Steam Workshop (since the dev branch seems to have fixed the issues with the latest update), is that ok? 
To be honest, even if you don't have access to a Windows machine, the process is so simple that I'm sure you guys should be able to add the main and dev versions **officially**, the main.js file is just `Game.LoadMod("https://cookiemonsterteam.github.io/CookieMonster/dist/CookieMonsterDev.js");` and all seems to be working fine from what I've tested.

In any case, if that's an issue, please let me know and I'll remove it from the workshop.

Here's the link: https://steamcommunity.com/sharedfiles/filedetails/?id=2820988936.

**Comment by DanielNoord** [comment]
> @DanielNoord Hey, I added the dev version of Cookie Monster to the Steam Workshop (since the dev branch seems to have fixed the issues with the latest update), is that ok? To be honest, even if you don't have access to a Windows machine, the process is so simple that I'm sure you guys should be able to add the main and dev versions **officially**, the main.js file is just `Game.LoadMod("https://cookiemonsterteam.github.io/CookieMonster/dist/CookieMonsterDev.js");` and all seems to be working fine from what I've tested.
> 
> In any case, if that's an issue, please let me know and I'll remove it from the workshop.
> 
> Here's the link: https://steamcommunity.com/sharedfiles/filedetails/?id=2820988936.

No, I'm not really looking to upload this myself. So thanks for the initiative.

Only thing I would ask you is if you could a line along the lines of:
`Development of Cookie Monster happens over at .... Community contributions to improve functionality or fix open bugs is much appreciated!`.

**Comment by gusluchetti** [comment]
Will do! Thanks for the quick reply.

**Comment by 3plus4i** [comment]
> @3plus4i That is indeed true! However, as long as the steam version is Windows-only I can't do anything to support it myself as I can't test anything.

Btw, steam works well for me under Xubuntu, with Proton and the Steam Play beta. It needs quite a lot of resources though and experiences seem to be mixed between Linux players. Meant to say that for months now.
